src/cgame/cg_rocket_draw.cpp

```diff
@@ -2174,6 +2174,131 @@ class PredictedMineEfficiencyElement : public HudElement
 	std::unordered_map<int, std::string> pluralSuffix;
 };
 
+class BarbsHudElement : public HudElement
+{
+public:
+	BarbsHudElement ( const Rocket::Core::String& tag ) :
+	HudElement ( tag, ELEMENT_ALIENS ),
+	numBarbs( 0 ),
+	maxBarbs( 0 ),
+	regenerationInterval ( 0 ),
+	t0 ( 0 ),
+	offset ( 0 ) {}
+
+	void OnAttributeChange( const Rocket::Core::AttributeNameList& changed_attributes )
+	{
+		HudElement::OnAttributeChange( changed_attributes );
+		if ( changed_attributes.find( "src" ) != changed_attributes.end() )
+		{
+			maxBarbs = BG_Weapon( WP_ALEVEL3_UPG )->maxAmmo;
+			if ( maxBarbs > 0 )
+			{
+				src = GetAttribute<Rocket::Core::String>( "src", "" );
+				Rocket::Core::String base( va("<img class='barbs' src='%s' />", src.CString() ) );
+				Rocket::Core::String rml;
+
+				for ( int i = 0; i < maxBarbs; i++ )
+				{
+					rml += base;
+				}
+
+				// SetInnerRML deletes all child Elements => bring children up to date
+				children.clear();
+				SetInnerRML( rml );
+				for( int i = 0; i < maxBarbs; i++) {
+					children.push_back( GetChild( i ) );
```

Comment:
Why do you create a copy of the children? To me it seems like you could just iterate over the superclasses children directly all the time, assuming that `GetChild` accesses them. (Note that I have no idea what these children are in the first place.)